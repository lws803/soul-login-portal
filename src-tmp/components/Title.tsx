import { Box, Text } from '@chakra-ui/react';

export default function Title({ title, subTitle }: Props) {
  return (
    <Box marginBottom="40px">
      <Text fontSize="4xl" fontWeight="bold">
        {title}
      </Text>
      {subTitle}
    </Box>
  );
}

type Props = {
  title: string;
  subTitle?: React.ReactNode;
};
