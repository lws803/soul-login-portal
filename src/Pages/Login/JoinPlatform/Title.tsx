import { CheckCircleIcon } from '@chakra-ui/icons';
import { HStack, Tooltip, Text } from '@chakra-ui/react';
import { Platform } from '../api';

export default function Title({ platform }: Props) {
  return (
    <HStack justifyContent="center" w={['90vw', '90vw', '500px', '500px']}>
      <Text
        fontSize="3xl"
        whiteSpace="nowrap"
        overflow="hidden"
        textOverflow="ellipsis"
        fontWeight="bold"
        flexShrink={1}
      >
        {platform.name}
      </Text>
      {platform.isVerified && (
        <Tooltip label="This platform is verified" flexShrink={0}>
          <span>
            <CheckCircleIcon
              color="soul.green.200"
              aria-label="Verified marker"
            />
          </span>
        </Tooltip>
      )}
    </HStack>
  );
}

type Props = {
  platform: Platform;
};
