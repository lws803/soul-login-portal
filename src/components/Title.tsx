import { Box, Text } from '@chakra-ui/react';

export default function Title({ title, subTitle }: Props) {
  return (
    <Box marginBottom="32px">
      <Text fontSize="4xl" fontWeight="bold">
        {title}
      </Text>
      <Box mt="8px">{subTitle}</Box>
    </Box>
  );
}

type Props = {
  title: string;
  subTitle?: React.ReactNode;
};
