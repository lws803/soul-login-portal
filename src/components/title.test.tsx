import { render } from '@testing-library/react';

import Title from './title';

describe(Title, () => {
  it('renders', () => {
    const { container } = render(<Title title="Title" subTitle="Sub title" />);
    expect(container).toMatchSnapshot();
  });
});
