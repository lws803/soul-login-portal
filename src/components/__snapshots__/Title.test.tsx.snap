// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders 1`] = `
<div>
  <div
    class="css-1yjvs5a"
  >
    <p
      class="chakra-text css-odt1ir"
    >
      Title
    </p>
    <div
      class="css-164r41r"
    >
      Sub title
    </div>
  </div>
</div>
`;
